first commit

ntretyakov1 opened the pull request.

ntretyakov1 committed:
first commit

KushchevR commented on README.MD:
```diff
@@ -0,0 +1,30 @@
+        - logo.png 130px x 100px;
+        - logo_medium.png 240px x 84px;
+        - logo_min.png 84px x 84px;
+        - logo_dp.png 174px x 109px;
```
Не вижу описание задачи, что необходимо выполнить?! Изменить!

KushchevR commented on api.js:
```diff
@@ -0,0 +1,283 @@
+		return axios
+			.get(
+				`${ROOT_PATH}/api/v4/leads/pipelines/${pipelineId}/statuses/${statusId}`,
+				{
+					headers: {
+						Authorization: `Bearer ${access_token}`,
+					},
+				}
+			)
+			.then((res) => res.data);
+	});
+
+	// Получить данные пользователя crm по его id
+	this.getUser = authChecker((userId) => {
+		return axios
+			.get(`${ROOT_PATH}/api/v4/users/${userId}`, {
+				headers: {
+					Authorization: `Bearer ${access_token}`,
+				},
+			})
+			.then((res) => res.data);
+	});
+
+	// Получить данные всех пользователей crm
+	this.getUsers = authChecker(({ page = 1, limit = LIMIT }) => {
+		const url = `${ROOT_PATH}/api/v4/users?${querystring.stringify({
+			page,
+			limit,
+		})}`;
+		return axios
+			.get(url, {
+				headers: {
+					Authorization: `Bearer ${access_token}`,
+				},
+			})
+			.then((res) => {
+				return res.data ? res.data._embedded.users : [];
+			});
+	});
+}
```
Слишком много не используемого, оставить только те методы, которые нужны

KushchevR commented on index.js:
```diff
@@ -0,0 +1,52 @@
+
+const app = express();
+
+const idFieldBirthDay = "375847";
+const idFieldAge = "377623";
```
нейминг в соответствии с правилами переписать (смотри константы)

KushchevR commented on index.js:
```diff
@@ -0,0 +1,52 @@
+app.use(express.json());
+app.use(express.urlencoded({ extended: true }));
+
+api.getAccessToken().then(() => {
```
Избавиться от цепочки. Вынести эндпоинт и внутри уже после срабатывания тригера вызывать логику

KushchevR commented on index.js:
```diff
@@ -0,0 +1,52 @@
+
+		const makeFieldAge = makeField(+idFieldAge, age);
+
+		const obj = {
```
Нейминг. Что за obj?! Непонятно

KushchevR commented on utils.js:
```diff
@@ -0,0 +1,141 @@
+	bulkOperation,
+	getAllPages,
+	getClearPhoneNumber
+};
```
Так же почисти. Оставить только то что используешь

KushchevR commented on widget/manifest.json:
```diff
@@ -0,0 +1,69 @@
+    "settings",
+    "digital_pipeline",
+    "advanced_settings",
+    "salesbot_designer"
```
Почитай доку "АМО" и оставь только там где реально используется виджет

KushchevR reviewed (changes requested):
В целом нужно почистить коммит от тех файлов, которые не нужны

ntretyakov1 committed:
refactor:
1.Почистил коммит от ненужных файлов;
2.В api.js оставил только те методы,которые используются;
3.В utils.js  оставил только те методы,которые используются;
4.В index.js поправил нейминг констант;
5.В index.js избавился от цепочки функций;
fix:
1.функция "сalculateAge" не будет возвращать отрицательные числа.
docs:
1.Добавил описание задачи в readme.

mzudin-reon commented on index.js:
```diff
@@ -0,0 +1,53 @@
+const api = require("./api");
+const logger = require("./logger");
+const config = require("./config");
+const { getFieldValue, makeField } = require("./utils");
```
Переписать на import

mzudin-reon commented on index.js:
```diff
@@ -0,0 +1,53 @@
+		return 0;
+	}
+	return age;
+};
```
вынести в utils

mzudin-reon commented on index.js:
```diff
@@ -0,0 +1,53 @@
+const app = express();
+
+const ID_FIELD_BIRTHDAY = "375847";
+const ID_FIELD_AGE = "377623";
```
создать файл consts.js и там хранить все константы

mzudin-reon commented on index.js:
```diff
@@ -0,0 +1,53 @@
+app.use(express.urlencoded({ extended: true }));
+
+app.post("/hook", (req, res) => {
+	const contact = req.body.contacts.add[0];
```
Избавиться от индекса
const [contact] = req.body.contacts.add;

mzudin-reon commented on index.js:
```diff
@@ -0,0 +1,53 @@
+	const birthdayValue = getFieldValue(contact.custom_fields, ID_FIELD_BIRTHDAY);
+
+	if (!birthdayValue) {
+		res.send("Отсутствует дата рождения");
+		return;
```
помести res.send в return

mzudin-reon commented on index.js:
```diff
@@ -0,0 +1,53 @@
+
+const сalculateAge = (year) => {
+	const age = new Date().getFullYear() - year;
+	if (Math.sign(age) === -1) {
```
Мне кажется if(age < 0) проще и понятнее

mzudin-reon commented on index.js:
```diff
@@ -0,0 +1,53 @@
+		custom_fields_values: [makeFieldAge],
+	};
+
+	api.updateContacts(contactForUpdate).then(() => {
```
api.updateContacts(contactForUpdate).then( () => res.send("OK") );

ntretyakov1 committed:
refactor:
1.Вынес в файл utils.js функцию "сalculateAge";
2.создал файл consts.js,там все константы.
3.Деструктуризация контакта в index.js.
4.Переместил все вызовы res.send() в return функций.

ntretyakov1 committed:
fix:
1.функция "сalculateAge" теперь учитывает день и месяц а не только год.
docs:
1.Поправил readme.md;
